legacy/mystery_game_nightmare27.py

```diff
@@ -0,0 +1,130 @@
+            self.guessed_letters.append(player_guess)
+            guess_result = self.calculate_new_word_list(player_guess)
+            if guess_result:
+                return 1
+            else:
+                self.remaining_guesses -= 1
+                return 0
+
+    def check_win_lose(self):
+        """ Checks to see if the player has won or lost the game """
+
+        if not self.remaining_guesses:
+            self.word = random.choice(self.possible_words)
+            return 1
+        elif "_" not in self.evasive_word:
+            return 2
+        return 0
+
+    def choose_word_class(self, class_list):
+        """Determines which word class gives the computer the highest chance
+        to win"""
+        best_class = "default"
+
+        for word_class in class_list:
+            if len(class_list[word_class]) > len(class_list[best_class]):
+                best_class = word_class
+
+        self.possible_words = class_list[best_class]
+
+        return best_class
+
+    def determine_class_types(self, player_guess):
+        """Examines words in list, returns class types based on location of
+        occurences of the player's guess in the string"""
+        class_type_list = {"default": []}
+
+        for word in self.possible_words:
+            class_title = ""
+            for letter_count in range(self.word_length):
+                if word[letter_count] == player_guess:
+                    class_title += "[" + str(letter_count) + "]"
+
+            if not len(class_title):
+                class_title = "None"
+
+            if class_title in class_type_list:
+                class_type_list[class_title].append(word)
+            else:
+                class_type_list[class_title] = [word]
+
+        return class_type_list
+
+    def load_dictionary(self, file):
+        """Loads the dictionary file from the specified location, file should
+        be in the format one word per line in the file"""
+        try:
+            with open(file, "r") as dictionary_file:
+                word_list = dictionary_file.read().split()
+            return word_list
+        except IOError:
+            file = self.default_dictionary
+            return self.load_dictionary(file)
+
+    def update_evasive_word(self, class_title, guess):
+        """Updates the evasive word to display any letters that are in the
+        word class that the computer player has chosen."""
+        parse_string = re.compile("(?<=\[)(\d+)(?=])")
+
+        indicies = re.findall(parse_string, class_title)
+
+        for index in indicies:
+            self.evasive_word[int(index)] = guess
+
+if __name__ == '__main__':
+    game = MysteryGameNightmare(file="dictionary.txt")
+    while True:
+        print(game.evasive_word)
+        print("{} Words left in pool".format(len(game.possible_words)))
+        guess = raw_input("Guess! : ")
+        if guess == "QUIT":
+            break
+        else:
+            game.check_guess(guess.upper())
```

Comment:
> +
> +        indicies = re.findall(parse_string, class_title)
> +
> +        for index in indicies:
> +            self.evasive_word[int(index)] = guess
> +
> +if __name__ == '__main__':
> +    game = MysteryGameNightmare(file="dictionary.txt")
> +    while True:
> +        print(game.evasive_word)
> +        print("{} Words left in pool".format(len(game.possible_words)))
> +        guess = raw_input("Guess! : ")
> +        if guess == "QUIT":
> +            break
> +        else:
> +            game.check_guess(guess.upper())

I meant to take this out.  The number that returns was originally used by
the interface to determine if the guess was correct, if the guess was
incorrect, if the guess was already in the list of guessed letters, or if
the guess was an invalid guess (number or string longer than one
character).  The if __name__ = ‘__main__’ code was just something I put in
to test the evil mode with infinite guesses so that I could observe that
the word pool was shrinking.  The class was only intended to be used by the
play_mystery_word.py script.  Since I knew I would always give correct
input and I wasn’t keeping track of number of guesses I just tossed the
return value.
Thanks for the feedback!  I was originally going to write the evil version
of the game as an extension of the original version, but when I realized I
was overriding almost every method ( I think the load_dictionary method was
the only thing that was exactly the same), I split it into two.  If I get
some time between projects I’ll try to come back and see how I can combine
them.

Also, as far as writing code that is compatible with both Python 3 and 2,
is there an easy way to set my sandbox up so that I can have both versions
available so that I could just run python <file> for 3 and python2 or
something similar to run my scripts in python 2?  I’d really like to play
around with PyGame some more and that would be really useful.  I really
would have preferred not to have the legacy folder at all, but I couldn’t
get PyGame to recognize my keyboard input in 3.  Should I just have put the
PyGame interface in the legacy folder and imported the game classes from
the mystery-word directory?  The only code in the original that gave me a
problem was catching the FileNotFoundError in the load dictionary method.
I had to change it to IOError to be compatible with 2.7.  I could easily
have just used IOError in the original version as well.

I’ll search around google as well to see if I can find the answer.


Thanks again!

Alan


On Jan 22, 2015, at 9:42 PM, Clinton N. Dreisbach <notifications@github.com>
wrote:

Closed #8.

—
Reply to this email directly or view it on GitHub.

On Thu, Jan 22, 2015 at 9:36 PM, Clinton N. Dreisbach <
notifications@github.com> wrote:

> In legacy/mystery_game_nightmare27.py
> <https://github.com/tiyd-python-2015-01/mystery-word/pull/8#discussion_r23428641>
> :
>
> > +
> > +        indicies = re.findall(parse_string, class_title)
> > +
> > +        for index in indicies:
> > +            self.evasive_word[int(index)] = guess
> > +
> > +if __name__ == '__main__':
> > +    game = MysteryGameNightmare(file="dictionary.txt")
> > +    while True:
> > +        print(game.evasive_word)
> > +        print("{} Words left in pool".format(len(game.possible_words)))
> > +        guess = raw_input("Guess! : ")
> > +        if guess == "QUIT":
> > +            break
> > +        else:
> > +            game.check_guess(guess.upper())
>
> Not sure what's happening here. check_guess returns a number, but you
> aren't using it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tiyd-python-2015-01/mystery-word/pull/8/files#r23428641>
> .
>